rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+        properties:
+          type: object
+          description: Configured string to string map of properties for the namespace
+          example: '{ "owner": "Hank Bendickson" }'
+          default: '{ }'
+
+    Namespace:
+      description: Reference to one or more levels of a namespace
+      type: array
+      items:
+        type: string
+      example: [ "accounting", "tax" ]
+
+    RenameTableRequest:
+      type: object
+      properties:
+        sourceTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        destinationTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+        - name
+      properties:
+        namespace:
+          type: array
+          description: Individual levels of the namespace
+          items:
+            type: string
+        name:
+          type: string
+          nullable: false
+
+    UpdatePropertiesRequest:
+      type: object
+      properties:
+        removables:
+          type: array
+          uniqueItems: true
+          items:
+            type: string
+          example: '[ "department", "access_group" ]'
+        updates:
+          uniqueItems: true
+          type: object
+          items:
+            type: string
+          example: { "owner": "Hank Bendickson" }
+
+
+  #############################
+  # Reusable Response Objects #
+  #############################
+  responses:
+
+    404Response:
+      description: Generic example response to demonstrate a 404's general form.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ResponseErrorObject'
+          example: {
+            "error": {
+              "message": "The given namespace does not exist",
+              "type": "NoSuchNamespaceException",
+              "code": 404
+            }
+          }
+
+    IcebergResponseObject:
+      description: JSON wrapper for all response bodies, with a data object and / or an error object
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              data:
+                $ref: '#/components/schemas/ResponseDataObject'
+              error:
+                $ref: '#/components/schemas/ResponseErrorObject'
+          example: { "data": { ... }, "error": { } }
+
+    DropTableResponse:
+      description: A successful call to drop a table
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              dropped:
+                type: boolean
+                description: true if the table was found and removed from the catalog
+              purged:
+                type: boolean
+                description: whether the underlying data was purged or is being purged
+
+    GetNamespaceResponse:
+      description: Returns a namespace and the properties stored on it
```

Comment:
This should probably mention that if properties are not supported the properties are optional? Or maybe they must be sent but can be empty?

Reply:
How I have it currently, on the response, it's defaulted to `{ }`. As within the Iceberg code, we don't really have the concept of a Namespace without properties. It's just implicit in some places that it's not supported.

I'll throw the words optional or when supported on there though to be more precise.

Reply:
Updated. I also indicated that `namespaces` is a required field, as well as gave a default value of an empty object to `properties` and marked is as the equivalent of `@Nullable`.

When the `properties` field is null, it indicates the server doesn't support namespace properties. If it returns an empty object, then there are simply no properties presently stored on this namesapce.

Reply:
We don't have to keep that way of distinguishing `null` vs `{}`, especially as there's no way to describe it in the OpenAPI document that I'm aware of, but it seemed like the easiest way to indicate truly empty vs not supported (and thus empty).